### ISSUE TITLE
Plop throwing SyntaxError from node-plop

### ISSUE THREAD

**Issue by tpizza** [amwmedia/plop#52]
Alright, so first off, I'm pretty stoked to be taking Plop for a spin.  It seems super promising and can't wait to roll it into my workflow.

It seem as though node-plop is breaking the CLI tool at the moment.  Plop is installed both as a global and a dev dependency for my project.  However, Plop fails when I try invoking it.  I've tested my own plopfile as well as pulled the one from the example dir and also found another in a gist which was tried.  All of them yield the same issue.

Here is the stack trace for the issue:

```
~\AppData\Roaming\npm\node_modules\plop\node_modules\node-plop\lib\modules\node-plop.js:29
function nodePlop(plopfilePath = '', plopCfg = {}) {
                               ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (~\AppData\Roaming\npm\node_modules\plop\node_modules\node-plop\lib\index.js:5:17)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)

```

**Comment by f-roland** [comment]
Saw that too this morning.
you need node > 6 to have support for default function argument.

try to update your node version, and plop should work. But maybe then add a warning when installing plop that it requires node > 6 ?

**Comment by amwmedia** [comment]
Thanks for letting me know, I'll get a hot fix out very soon. My intent is to support node 4.0+ so this is just a defect. I wonder why the babel compile process doesn't transform this... very strange

**Comment by f-roland** [comment]
Well, it is working, but requires a certain node version.
if your intent is to support node 4+, maybe you could add it in package.json ?
https://docs.npmjs.com/files/package.json#engines

**Comment by amwmedia** [comment]
I added engines to node-plop, but I should probably add that to plop as well. Thanks for the suggestion. I think what I need is a CI process that just runs all the tests on node 4

**Comment by tpizza** [comment]
@amwmedia so sort of weird/interesting...I forked node-plop and when compiling, Babel transformed the default parameters properly.  

The only other problem that happened thereafter was the `path.posix.basename` call for setting the plopfile path was triggering an error because the posix basename was returning the full path for whatever reason.  I changed the call to `path.basename` and now plop is working for me properly.

I can submit that as a PR if you'd like.  Although, it still doesn't solve the transpiler issue when the plugin's installed...Maybe because it's currently pulling 0.5.0 instead of 0.5.1?
